lms/djangoapps/course_blocks/transformers/library_content.py

```diff
@@ -0,0 +1,176 @@
+"""
+Content Library Transformer.
+"""
+import json
+from courseware.models import StudentModule
+from openedx.core.lib.block_cache.transformer import BlockStructureTransformer
+from xmodule.library_content_module import LibraryContentModule
+from xmodule.modulestore.django import modulestore
+from eventtracking import tracker
+
+
+class ContentLibraryTransformer(BlockStructureTransformer):
+    """
+    A transformer that manipulates the block structure by removing all
+    blocks within a library_content module that a user should not have
+    access.
+
+    Staff users are *not* exempted from library content pathways.
+    """
+    VERSION = 1
+
+    @classmethod
+    def name(cls):
+        """
+        Unique identifier for the transformer's class;
+        same identifier used in setup.py.
+        """
+        return "library_content"
+
+    @classmethod
+    def collect(cls, block_structure):
+        """
+        Collects any information that's necessary to execute this
+        transformer's transform method.
+        """
+        block_structure.request_xblock_fields('mode')
+        block_structure.request_xblock_fields('max_count')
+        block_structure.request_xblock_fields('category')
+        store = modulestore()
+
+        # needed for analytics purposes
+        def summarize_block(usage_key):
+            """ Basic information about the given block """
+            orig_key, orig_version = store.get_block_original_usage(usage_key)
+            return {
+                "usage_key": unicode(usage_key),
+                "original_usage_key": unicode(orig_key) if orig_key else None,
+                "original_usage_version": unicode(orig_version) if orig_version else None,
+            }
+
+        # For each block check if block is library_content.
+        # If library_content add children array to content_library_children field
+        for block_key in block_structure.topological_traversal(
```

Comment:
This is a style choice you should feel free to disagree with, but I'd define an intermediate variable for the interator. The way it is makes the indentation harder to parse